go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/notaryproject/notation-core-go
+
+go 1.18
```

Comment:
I want someone with more familiarity with golang chime in here. Currently, notation and notation-go are on 1.17. Reading 1.18 [release notes](https://go.dev/doc/go1.18), it looks like 1.18 is backward compatible and it is also fixing some bugs present in earlier Go versions. Let me know if there a reason we shouldn't upgrade to 1.18.

Reply:
@shizhMSFT , @qmuntal, any feedback here?

Reply:
@shizhMSFT , @qmuntal, any feedback here?

Reply:
The `go` directive inside `go.mod` does not refer to the go version that will be used to compile the library, but to the maximum set of language features that this library supports. For example, generics are only available if we set it to `1.18` and `embed` if we set it to `1.16` or greater.

A customer using a lower Go toolchain version can still use a module whose `go` directive if set to a higher version unless the module really uses a newer language feature.

The best practice is to set the `go` directive to the lowest supported Go version. As we are not using generics here, I would set it to `1.17` to be in-line with `notation-go`.

Reply:
Good to know. Updated the PR.

@gokarnm Can you merge if this looks good?